src/models/CommonModel.ts

```diff
@@ -82,6 +82,23 @@ export class CommonModel extends CommonSchema<CommonModel> {
     return this.required.includes(propertyName);
   }
 
+  /**
+   * Adds an item to the model.
+   * 
+   * If items already exist the two are merged.
+   * 
+   * @param itemModel 
+   * @param schema schema to the corresponding property model
+   */
+  addItem(itemModel: CommonModel, schema: Schema) {
+    if (this.items !== undefined) {
+      Logger.warn(`While trying to add item to model ${this.$id}, duplicate items found. Merging models together to form a unified item model.`, itemModel, schema, this);
+      this.items = CommonModel.mergeCommonModels(this.items as CommonModel, itemModel, schema);
```

Comment:
Only reminder, we can have tuple type 😄

Reply:
Yep, on the milestone list will be changed afterwards - #67